### PR TITLE
Add common signalfx host id extraction

### PR DESCRIPTION
**Pull request by jrcamp**
This adds a shared way to extract AWSUniqueId and gcp_id dimensions. It will be
used by both sapmexporter and signalfxexporter. signalfxexporter cannot
currently use the method because it's not on pdata and it seemed silly to
unconvert metrics data back to pdata. I've added a note to signalfxexporter to
use this common implementation once it switches to pdata.

### CODE REVIEW

File: exporter/signalfxexporter/translation/converter.go
Comment (keitwb) [inline exporter/signalfxexporter/translation/converter.go:419]: Are you going to still remove the resource attributes used to derive the host id like the code below in this function does?  If so, it seems like there will need to be something more that the hostid module does to facilitate that instead of having to know the exact fields here.

Comment (jrcamp) [reply exporter/signalfxexporter/translation/converter.go:419]: Nope, sapm won't remove them since for traces we want all those tags. I think signalfxexporter will probably need to maintain its own list of labels/dimensions to filter out. It doesn't need to be per-provider type like it is today, it can just delete all of them (provider, account id, region, zone, etc. regardless of provider type).

**Review by keitwb — APPROVED** [review]

File: internal/common/splunk/hostid.go
Comment (dmitryax) [inline internal/common/splunk/hostid.go:63]: resourcedetection processor reports "aws" and "gcp", not "ec2" and "gce". I fixed that in signalfx exporter https://github.com/open-telemetry/opentelemetry-collector-contrib/commit/3cc0d376e9727a4cead5dbf96a50fd09e4ce9da4

File: internal/common/splunk/hostid.go
Comment (dmitryax) [inline internal/common/splunk/hostid.go:29]: As public constants, would it be better to make them more descriptive like `HostIDKeyAWS`, `HostIDKeyGCP`?

**Review by dmitryax — APPROVED** [review]

File: internal/common/cloud/providers.go
Comment (tigrannajaryan) [inline internal/common/cloud/providers.go:21]: Should this be in core semantic conventions or this is Splunk convention only?

File: internal/common/splunk/hostid.go
Comment (tigrannajaryan) [inline internal/common/splunk/hostid.go:44]: Name in the comment does not match the func name.

File: internal/common/splunk/hostid.go
Comment (tigrannajaryan) [inline internal/common/splunk/hostid.go:47]: Is this used anywhere? I cannot find. Is the intent that it will be used in a future PR?

File: internal/common/cloud/providers.go
Comment (jrcamp) [reply internal/common/cloud/providers.go:21]: aws and gcp are called out as examples in the conventions (https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/resource/semantic_conventions/cloud.md). Should I create a PR to amend semantic conventions to make it explicit that these values should be used for the platforms? If so I think then it'd make sense to put this in translator/conventions/opentelemetry.go.

File: internal/common/splunk/hostid.go
Comment (jrcamp) [reply internal/common/splunk/hostid.go:47]: Not yet, will be in future PR (APM correlation).

File: internal/common/cloud/providers.go
Comment (tigrannajaryan) [reply internal/common/cloud/providers.go:21]: Yes, best to add to core conventions.

**Review by tigrannajaryan — APPROVED** [review]
LGTM, I can merge as is if you prefer and you can move conventions to core in separate PR.

**Comment by jrcamp** [comment]
Ok let's do that, will open PR to spec to add to conventions and update here once it lands.

**Comment by jrcamp** [comment]
> Ok let's do that, will open PR to spec to add to conventions and update here once it lands.

https://github.com/open-telemetry/opentelemetry-specification/pull/1018

**Comment by dmitryax** [comment]
@tigrannajaryan @bogdandrutu can we merge this? I have a PR which depends on it

**Pull request merged**
